### PR TITLE
Adding examples for URL methods

### LINKED ISSUE #3516: Adapter url for methods need examples
http://emberjs.com/api/data/classes/DS.RESTAdapter.html#method_urlForFindHasMany

### PR DESCRIPTION
**Pull request by kylecoberly**
Fixes #3516

### CODE REVIEW

**Comment by fivetanley** [comment]
awesome! If you could add the word javascript by the ` ``` `, that helps it get syntax highlighted on the website.

For example:

```markdown
```javascript
let foo = blah
```

Can you also prefix your commit message with `[DOC release]` ?

**Comment by fivetanley** [comment]
Travis is also complaining about some style stuff, but otherwise this is really great! Thank you!

**Comment by kylecoberly** [comment]
Updated! I think I got the trailing whitespace issue fixed too.

File: packages/ember-data/lib/adapters/build-url-mixin.js [hunk 1]
```diff
@@ -153,6 +159,9 @@ var BuildURLMixin = Ember.Mixin.create({
   urlForFindQuery: urlForFindQuery,
 
   /**
+   * ```javascript
+   * let url = this.urlForQuery({author: "Igor Terzic"}, "post"); // "http://www.example.com/posts"
```
Comment (wecc) [inline packages/ember-data/lib/adapters/build-url-mixin.js]: Nitpicking here but should be `{ author: "Igor Terzic" }`

File: packages/ember-data/lib/adapters/build-url-mixin.js [hunk 2]
```diff
@@ -167,6 +176,9 @@ var BuildURLMixin = Ember.Mixin.create({
   },
 
   /**
+   * ```javascript
+   * let url = this.urlForFindMany([42,43,44], "post"); // "http://www.example.com/posts"
```
Comment (wecc) [inline packages/ember-data/lib/adapters/build-url-mixin.js]: Whitespace here too, `[42, 43, 44]`

**Comment by kylecoberly** [comment]
Whitespace updated, commits squashed

**Comment by wecc** [comment]
Thank you!

**Comment by igorT** [comment]
Thanks!

**Pull request merged**